test/deploy/rackhd_source_install.py

```diff
@@ -34,197 +34,211 @@
-                                )
-            pdudirect_cfg.close()
-            # copy file to ORA
-            fit_common.scp_file_to_ora('pdudirect.cfg')
-            self.assertEqual(fit_common.remote_shell('cp pdudirect.cfg /etc/network/interfaces.d/')['exitcode'], 0, "DHCP Config failure.")
-            os.remove('pdudirect.cfg')
-            # startup NIC
-            fit_common.remote_shell('ip addr add 192.168.1.1/24 dev ' + ifslit[2])
-            fit_common.remote_shell('ip link set ' + ifslit[2] + ' up')
-            self.assertEqual(fit_common.remote_shell('ping -c 1 -w 5 192.168.1.1')['exitcode'], 0, 'PDU NIC failure.')
-
-        #create DHCP config
-        fit_common.remote_shell('echo INTERFACES=' + ifslit[1] + ' > /etc/default/isc-dhcp-server')
+            # if 'pdu' is specified in stack, then configure PDU network
+            if 'pdu' in fit_common.fitcfg():
+                # process PDU IP configuration
+                pdusplit = fit_common.fitcfg()['pdu'].split(".")
+                pdu_prefix = pdusplit[0] + '.' + pdusplit[1] + '.' + pdusplit[2] + '.'
+                # build interface config file
+                pdudirect_cfg = open('pdudirect.cfg', 'w')
+                pdudirect_cfg.write('auto ' + ifslist[2] + '\n'
+                                    'iface ' + ifslist[2] + ' inet static\n'
+                                    'address ' + pdu_prefix + '1\n'
+                                    'netmask 255.255.255.0\n')
+                pdudirect_cfg.close()
+                # copy file to Host
+                fit_common.scp_file_to_host('pdudirect.cfg')
+                self.assertEqual(fit_common.remote_shell('cp pdudirect.cfg /etc/network/interfaces.d/')
+                                 ['exitcode'], 0, "DHCP config failure.")
+                os.remove('pdudirect.cfg')
+                # startup NIC
+                fit_common.remote_shell('ip addr add ' + pdu_prefix + '1/24 dev ' + ifslist[2])
+                fit_common.remote_shell('ip link set ' + ifslist[2] + ' up')
+                self.assertEqual(fit_common.remote_shell('ping -c 1 -w 5 ' + pdu_prefix + '1')
+                                 ['exitcode'], 0, 'PDU NIC failure.')
+            else:
+                log.info_5("**** No PDU specified for this stack")
+
+        # create DHCP config
+        fit_common.remote_shell('echo INTERFACES=' + ifslist[1] + ' > /etc/default/isc-dhcp-server')
+        # calculate control LAN IP configuration
+        ipsplit = fit_common.fitrackhd()['dhcpGateway'].split(".")
+        ip_prefix = ipsplit[0] + '.' + ipsplit[1] + '.' + ipsplit[2] + '.'
+        masksplit = fit_common.fitrackhd()['dhcpSubnetMask'].split(".")
+        dhcp_high = \
+            str(int(ipsplit[0]) + (255 - int(masksplit[0]))) + '.' + \
+            str(int(ipsplit[1]) + (255 - int(masksplit[1]))) + '.' + \
+            str(int(ipsplit[2]) + (255 - int(masksplit[2]))) + '.' + '254'
```

Comment:
For what it's worth, monorail.py uses the socket library to validate ip addresses.  It uses  socket.inet_aton(ip) to "convert an IPv4 address from dotted-quad string format (for example, ‘123.45.67.89’) to 32-bit packed binary format, as a string four characters in length".  It will generate a  "socket.error" if something is incorrect.